src/app/hooks/useViewTracker/index.test.jsx

```diff
@@ -0,0 +1,182 @@
+/* eslint-disable no-template-curly-in-string */
+import React from 'react';
+import { renderHook, act } from '@testing-library/react-hooks';
+import { useInView } from 'react-intersection-observer';
+
+import { ServiceContextProvider } from '#contexts/ServiceContext';
+import { RequestContextProvider } from '#contexts/RequestContext';
+import * as atiUrl from '#containers/ATIAnalytics/atiUrl';
+import useViewTracker from '.';
+
+import { CORRESPONDENT_STORY_PAGE } from '#app/routes/utils/pageTypes';
+import pageData from './pageData.json';
+
+delete window.location;
+window.location = { href: 'http://bbc.com/pidgin/tori-51745682' };
+process.env.SIMORGH_ATI_BASE_URL = 'https://logws1363.ati-host.net?';
+
+jest.mock('react-intersection-observer');
+const wait = duration => new Promise(resolve => setTimeout(resolve, duration));
```

Comment:
Using a real timeout in tests can significantly extend the run time of the test suite, it's worth thinking about mocking `setTimeout` itself for the purpose of the test. Afaik our tests run in series rather than parelell so this makes this problem particularly pronounced.

Reply:
Yeh I was having trouble with jest.useFakeTimers - just couldn't get it working - used a wait for now to get things going but will definitely either mock setTimeout or get jest.useFakeTimers working.